spyder/plugins/completion/tests/conftest.py

```diff
@@ -73,85 +70,31 @@ def reset_spyder(self):
         pass
 
 
-def register_all_providers():
-    """Create a entry points distribution to register all the providers."""
-    # This is not necessary in CIs
-    if running_in_ci():
-        return
-
-    fallback = pkg_resources.EntryPoint.parse(
-        'fallback = spyder.plugins.completion.providers.fallback.provider:'
-        'FallbackProvider'
-    )
-    snippets = pkg_resources.EntryPoint.parse(
-        'snippets = spyder.plugins.completion.providers.snippets.provider:'
-        'SnippetsProvider'
-    )
-    lsp = pkg_resources.EntryPoint.parse(
-        'lsp = spyder.plugins.completion.providers.languageserver.provider:'
-        'LanguageServerProvider'
-    )
-
-    # Create a fake Spyder distribution
-    d = pkg_resources.Distribution(__file__)
-
-    # Add the providers to the fake EntryPoint
-    d._ep_map = {
-        'spyder.completions': {
-            'fallback': fallback,
-            'snippets': snippets,
-            'lsp': lsp
-        }
-    }
-    # Add the fake distribution to the global working_set
-    pkg_resources.working_set.add(d, 'spyder')
-
-
-def remove_fake_distribution():
-    """Remove fake entry points from pkg_resources"""
-    # This is not necessary in CIs
-    if running_in_ci():
-        return
-
-    pkg_resources.working_set.by_key.pop('unknown')
-    pkg_resources.working_set.entry_keys.pop('spyder')
-    pkg_resources.working_set.entry_keys.pop(__file__)
-    pkg_resources.working_set.entries.remove('spyder')
-
-
 @pytest.fixture(scope="module")
 def qtbot_module(qapp, request):
     """Module fixture for qtbot."""
     result = QtBot(request)
     return result
 
 
-def create_completion_plugin(create_fake_distribution=None,
-                             remove_fake_distribution=None):
+def create_completion_plugin():
     @pytest.fixture(scope='module')
     def completion_plugin_wrap(qtbot_module, request):
-        if create_fake_distribution is not None:
-            create_fake_distribution()
-
         main_window = MainWindowMock()
         completions = CompletionPlugin(main_window, CONF)
 
         # Remove Kite (In case it was registered via setup.py)
         completions.providers.pop('kite', None)
 
         def teardown():
-            if remove_fake_distribution is not None:
-                remove_fake_distribution()
+            pass
 
         request.addfinalizer(teardown)
         return completions
     return completion_plugin_wrap
 
 
-completion_plugin = create_completion_plugin()
-completion_plugin_all = create_completion_plugin(
-    create_fake_distribution=register_all_providers,
-    remove_fake_distribution=remove_fake_distribution)
+completion_plugin_all = create_completion_plugin()
```

Comment:
This is causing the tests to fail, because some tests still use `completion_plugin`, speficially `test_plugin.py::test_configuration_merge` (that one was the one that failed and the only obvious canidate I found by grepping, but there may be others); I suggest switching it/them over to use the `_all` variant.